web-wallet/src/lib/stores/walletStore.js

```diff
@@ -84,7 +109,7 @@ const stake = async (amount, gasSettings) =>
   asyncNoopFailure(amount, gasSettings);
 
 /** @type {WalletStoreServices["sync"]} */
-async function sync(/* from */) {
+async function sync(fromBlock) {
   const store = get(walletStore);
 
   if (!store.initialized) {
```

Comment:
Could we have this as:

```
if (syncPromise) {
  return syncPromise;
}
```

instead, so as to save one indentation. The method is quite long anyways, would be good to simplify it as much as we can.

Reply:
Nevermind, seems like we have to return a sync promise.